Tests/NIOTests/ChannelPipelineTest.swift

```diff
@@ -620,4 +620,48 @@ class ChannelPipelineTest: XCTestCase {
         XCTAssertTrue(try h1 === channel.pipeline.context(handlerType: TestHandler.self).wait().handler)
         XCTAssertFalse(try h2 === channel.pipeline.context(handlerType: TestHandler.self).wait().handler)
     }
+
+    func testContextForHeadOrTail() throws {
+        let channel = EmbeddedChannel()
+
+        defer {
+            XCTAssertFalse(try channel.finish())
+        }
+
+        do {
+            _ = try channel.pipeline.context(name: "head").wait()
```

Comment:
Might be better to look for these handlers by type, as we couldn't fix that by just removing their names.

Reply:
added an extra assert for that